euphoria-core/src/main/java/cz/seznam/euphoria/core/client/flow/Flow.java

```diff
@@ -335,6 +338,23 @@ public Settings getSettings() {
   }
 
   /**
+   * A convenience method to create a data set from the given source and
+   * assign the elements event time using the user defined function.
+   *
+   * @param <T> the type of elements of the created input data set
+   *
+   * @param source the data source to represent as a data set
+   * @param evtTimeFn the user defined event time extraction function
+   *
+   * @return a data set representing the specified source of data with assigned
+   *          event time assigned
+   */
+  public <T> Dataset<T> createInput(DataSource<T> source, ExtractEventTime<T> evtTimeFn) {
```

Comment:
:+1: this is really cool. When we discussed this I didn't quite realized that this form of time assignment is just a special case of the one we agreed to implement.